app/src/main/java/com/gruppone/stalker/HeadersAdders.java

```diff
@@ -1,15 +1,20 @@
 package com.gruppone.stalker;
 
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.android.volley.AuthFailureError;
 import com.android.volley.toolbox.JsonArrayRequest;
 import com.android.volley.toolbox.JsonObjectRequest;
 import java.util.HashMap;
 import java.util.Map;
 
+//XXX Since we now know that all request and response bodies are always JSON objects
+//    and never arrays, all this ordeal with a separate interface can be resolved
+//    by just using a Decorator pattern on the JsonObjectRequest type
 public abstract class HeadersAdders {
```

Comment:
questa la rinominerei o in HeadersAdder (classe "aggiungitore di headers") o HeaderAdders (metodi "aggiungitori di header")

Reply:
Questa interfaccia viene levata immediatamente. Serviva solo come taccone al fatto che avremmo dovuto usare il decorator pattern su due classi diverse, JsonObjectRequest e JsonArrayRequest, visto che alcune response del server ritornavano un array. Con la ristrutturazione delle api non si pone più il problema e possiamo usare un decorator come si deve su JsonObjectRequest.